cookbook/lvgl.rst

```diff
@@ -0,0 +1,1655 @@
+                        control:
+                          no_repeat: true
+                      - text: 2
+                        control:
+                          no_repeat: true
+                      - text: 3
+                        control:
+                          no_repeat: true
+                  - buttons:
+                      - text: 4
+                        control:
+                          no_repeat: true
+                      - text: 5
+                        control:
+                          no_repeat: true
+                      - text: 6
+                        control:
+                          no_repeat: true
+                  - buttons:
+                      - text: 7
+                        control:
+                          no_repeat: true
+                      - text: 8
+                        control:
+                          no_repeat: true
+                      - text: 9
+                        control:
+                          no_repeat: true
+                  - buttons:
+                      - text: "\uF55A"
+                        key_code: "*"
+                        control:
+                          no_repeat: true
+                      - text: 0
+                        control:
+                          no_repeat: true
+                      - text: "\uF00C"
+                        key_code: "#"
+                        control:
+                          no_repeat: true
+
+    key_collector:
+      - source_id: lvgl_keypad
+        min_length: 4
+        max_length: 4
+        end_keys: "#"
+        end_key_required: true
+        back_keys: "*"
+        allowed_keys: "0123456789*#"
+        timeout: 5s
+        on_progress:
+          - if:
+              condition:
+                lambda: return (0 != x.compare(std::string{""}));
+              then:
+                - lvgl.label.update:
+                    id: lvgl_label
+                    text: !lambda 'return x.c_str();'
+              else:
+                - lvgl.label.update:
+                    id: lvgl_label
+                    text: "Enter code and \uF00C"
+        on_result:
+          - if:
+              condition:
+                lambda: return (0 == x.compare(std::string{"1234"}));
+              then:
+                - lvgl.led.update:
+                    id: lvgl_led
+                    color: 0x00FF00
+              else:
+                - lvgl.led.update:
+                    id: lvgl_led
+                    color: 0xFF0000
+
+A few notable things in this example: usage of a base object ``obj`` as a parent for the label (in order to center the label in the middle of it and emphasize it with shadows independently of the label's dimensions); usage of ``align_to`` to align it to the led vertically; changing the background color of the buttons in ``pressed`` state; using the ``key_code`` configuration option to send a different character to ``key_collector`` instead of the displayed symbol.
```

Comment:
```suggestion
Of note:

- A base object ``obj`` is used as a parent for the label; this allows proper centering of the label as well as emphasizing it with shadows independently of the label's dimensions.
- ``align_to`` is used to align the label to the ``led`` vertically.
- Changing the background color of the buttons in ``pressed`` state.
- Use of the ``key_code`` configuration to send a different character to ``key_collector`` instead of the displayed symbol.
```